R/differential_colocalization_analysis.R

```diff
@@ -111,12 +154,17 @@ RunDCA.data.frame <- function (
     })
     # Export variables to each cluster
     clusterExport(cl, c("test_groups", "test_groups_keys", "contrast_column",
```

Comment:
Unfortunately yes. This created a lot of overhead when using a cluster. We might be able to workaround this somehow, but if we send chunks to each cluster we lose a lot of the flexibility that we have now.

Reply:
What flexibility do you mean? We could also consider using `multidplyr` and `do`, where you can partition chunks easily. But if this works it works 😄

Reply:
The way it's set up now, we send chunks if row IDs to compute tests for in parallel. Each chunk should be sent to a separate node/thread. Within the `pblapply`, we then use these IDs to fetch the relevant data from `test_groups` for the wilcox tests and additional data such as marker IDs from `test_groups_keys`. This means that both of these objects needs to be accessible within the environments created on each node.

With `clusterExport`, we can copy these objects to each node so that they are available and this is what I believe makes things a bit slow initially. However, once these objects are copied everything should run pretty fast.

I believe `multidplyr` suffers from the same issue. We can probably refactor the code to handle this better but it would be much easier to do this on a Windows machine.